plugins/woocommerce/src/Admin/Features/Blueprint/Init.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare( strict_types = 1);
+
+namespace Automattic\WooCommerce\Admin\Features\Blueprint;
+
+use Automattic\WooCommerce\Admin\PageController;
+
+/**
+ * Class Init
+ *
+ * This class initializes the Blueprint feature for WooCommerce.
+ */
+class Init {
+	/**
+	 * Init constructor.
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'init_rest_api' ) );
+		add_filter( 'woocommerce_admin_shared_settings', array( $this, 'add_upload_nonce_to_settings' ) );
```

Comment:
The Init class won't be loaded unless `blueprint` flag is enabled. Anything in the class is already feature checked as far as I know.

Reply:
@moon0326 Nice! I'm curious which part of the code that does this? I couldn't find a reference where it's conditionally imported

Reply:
@ilyasfoo So the features are loaded with [init](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/Features.php#L61) hook, which calls [load_features()](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/Features.php#L131) function.

load_features() loops through only [enabled](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/FeaturePlugin.php#L209) features and initializes the feature class.

The feature classname can be in two different format:

- [Automattic\\WooCommerce\\Admin\\Features\\FEATURE-NAME](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/Features.php#L116)

- [Automattic\\WooCommerce\\Admin\\Features\\FEATURE-NAME\\Init](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/Features.php#L121)

Reply:
Got it, thanks! I don't particularly like it, feels a bit like sorcery. But to be fair it's a known PHP pattern

Reply:
Thinking out loud, it might be better to opt for [FeaturesController](https://github.com/woocommerce/woocommerce/blob/657a3fb99acff25805f51c484e372105de11000e/plugins/woocommerce/src/Internal/Features/FeaturesController.php#L23) instead of feature flag. 🤔 

Though not something to be addressed in this PR

Reply:
@ilyasfoo Totally agree! I think we should start using FeaturesController from now on. Let's address it in a separate PR.